lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -359,22 +359,12 @@ def ready_image(action_handler, image_spec, image_options)
     end
 
     def destroy_image(action_handler, image_spec, image_options)
-      actual_image = image_for(image_spec)
-      if actual_image.nil? || !actual_image.exists?
-        Chef::Log.warn "Image #{image_spec.name} doesn't exist"
-      else
-        snapshots = actual_image.block_device_mappings.map do |dev, opts|
-          ec2.snapshots[opts[:snapshot_id]]
-        end
-        action_handler.perform_action "De-registering image #{image_spec.name}" do
-          actual_image.deregister
-        end
-        if snapshots.any?
-          action_handler.perform_action "Deleting image #{image_spec.name} snapshots" do
-            snapshots.each do |snap|
-              snap.delete
-            end
-          end
+      # TODO the driver should automatically be set by `inline_resource`
+      d = self
+      Provisioning.inline_resource(action_handler) do
```

Comment:
TODO try propogating the driver, the event queue, chef_provisioning, etc.

We could also call `with_driver`, `with_chef_server`, etc. at the beginning of the recipe.

Make `Provisioning.inline_resource` accept driver as an optional parameter.